dan_layer/common_types/src/num_preshards.rs

```diff
@@ -0,0 +1,78 @@
+//    Copyright 2024 The Tari Project
+//    SPDX-License-Identifier: BSD-3-Clause
+
+use std::{error::Error, fmt::Display};
+
+use serde::{Deserialize, Serialize};
+
+#[cfg_attr(
+    feature = "ts",
+    derive(ts_rs::TS),
+    ts(export, export_to = "../../bindings/src/types/")
+)]
+#[derive(Clone, Debug, Copy, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub enum NumPreshards {
+    One = 1,
+    Two = 2,
+    Four = 4,
+    Eight = 8,
+    Sixteen = 16,
+    ThirtyTwo = 32,
+    SixtyFour = 64,
+    OneTwentyEight = 128,
+    TwoFiftySix = 256,
```

Comment:
A bit of a nitpick, but I think using the actual numbers on the enum names (e.g. `P128`, `P256`) is more readable in this case

Reply:
Agree with that - Ok if I rename them in the next PR?